### PR TITLE
feature: add exclude_label parameter to list_pods method

### PR DESCRIPTION
**Pull request by priyansh-saxena1**
## Add exclude_label parameter to KrknKubernetes.list_pods()

This PR enhances the `list_pods()` method with a powerful new filtering capability: the `exclude_label` parameter. 

## Description
Added label exclusion to KrknKubernetes.list_pods() method. Users can now exclude pods matching specific labels using the same `key=value` format as label_selector. This enables simultaneous positive and negative filtering for precise pod selection.

## Technical Details
The implementation required:
- Adding the parameter to the method signature with a default value of `None` to maintain backward compatibility
- Modifying the pod iteration logic to check for label matches and skip those pods
- Adding comprehensive test coverage to verify all edge cases

## Testing 
I've implemented comprehensive tests that verify:
- Basic functionality works correctly with namespace filtering
- Pods without the exclude label are properly included
- Pods with matching exclude labels are correctly filtered out
- Edge cases are handled properly (null labels, etc.)

All tests pass successfully in both local and CI environments.

## Documentation
- [] **Is documentation needed for this update?**

If checked, a documentation PR must be created and merged in the [website repository](https://github.com/krkn-chaos/website/).

## Related Documentation PR

---

solves https://github.com/krkn-chaos/krkn/issues/784
references : https://github.com/krkn-chaos/krkn/pull/811#issuecomment-2901842181

### CODE REVIEW

**Review by jason810496 — COMMENTED** [review]
Would using `exclude_labels: list[str]` instead of a single key-value pair better support the customer use case? It would be great to have someone more familiar with the context take another look and share their thoughts.

I've left a few nits on the PR as well. Thanks!

File: src/krkn_lib/k8s/krkn_kubernetes.py [hunk 1]
```diff
@@ -550,6 +552,14 @@ def list_pods(
             raise e
         for ret_list in ret:
```
[the remark this answers is not in the document]
Comment (Transcendental-Programmer) [reply src/krkn_lib/k8s/krkn_kubernetes.py:553]: your suggestion to optimize by moving the split operation outside the nested loops is a good implementation. I'll update the implementation to reflect the same.

This avoids redundant parsing and simplifies the logic. Great suggestion, Thanks!

**Comment by Transcendental-Programmer** [comment]
> Would using `exclude_labels: list[str]` instead of a single key-value pair better support the customer use case? It would be great to have someone more familiar with the context take another look and share their thoughts.
> 
> I've left a few nits on the PR as well. Thanks!


1. it is more cleaner than running multiple filtered queries!
2. one can build their own exclusion list based on test conditions:

```
exclusions = []
if is_production:
    exclusions.append("tier=production")
if during_business_hours:
    exclusions.append("availability=high")
```


those were my thoughts as it would accomodate complex queries.

**Review by paigerube14 — APPROVED** [review]

**Review by tsebastiani — COMMENTED** [review]
**Context**: I've tested this PR on kind with `docker` cloud
While testing this PR I've found some issues not strictly related to it, but is better to keep track of them:
- docker lib needs to be updated to 7.1.0 
- needs a rebase because affected_nodes_status has not been injected [in the docker constructor](https://github.com/krkn-chaos/krkn/blame/d47286ae21ebbe5c224fa8d8ebc869d1f0377e49/krkn/scenario_plugins/node_actions/node_actions_scenario_plugin.py#L112)

Regarding this PR these are my observations:
- the scenario seems to not have effect on kind nodes
- the pod used to inject the scenario is not removed (that's the main problem)
